tests/environment.py

```diff
@@ -197,16 +197,23 @@ def _docker_wait_for_log(container: str, program: str, regex: str, start_line: i
 
 
 def _device_wait_for_log(device: None, log_path: str, regex: str,
-                         start_line: int, timeout: int) -> bool:
-    """Waits for log mathing regex expression to show up."""
-    device.sendline("tail -n10 -f {}".format(log_path))
-    match = device.expect(
-            pattern=[regex, device.pexpect.EOF, device.pexpect.TIMEOUT],
-            timeout=timeout)
-    if match == 1:
-        return True
+                         timeout_set: int, start_line: int = 0):
+    """Waits for log matching regex expression to show up."""
+    device.sendline("tail -f -n +{:d} {}".format(int(start_line)+int(1), log_path))
+    device.expect(regex, timeout=timeout_set)
+    match = device.match.group(0)
+    # Send Ctrl-C to interrupt tail -f
+    device.send('\003')
+    device.expect(device.prompt)
+    if match:
+        device.sendline("grep -n \"{}\" {}".format(match, log_path))
+        # Typical output of grep -n from log: "line_num:severity"
+        # this regex has to capture just number of line in log
+        device.expect(r"(?P<line_number>[0-9]+):[A-Z]+\s[0-9]", timeout=timeout_set)
+        matched_line = device.match.group('line_number')
+        return (True, matched_line, match)
     else:
-        return False
+        return (False, start_line, None)
```

Comment:
>format return as it's expected by callers.

There is something to be sorted out with _device_wait_for_log.
It returns a free-form tuple that is not mentioned anywhere in the documentation.
(this hasn't been introduced in this PR)